File: server/WebProjekt/src/main/java/com/example/service/ProductService.java
package com.example.service;

import com.example.entity.Product;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.concurrent.ExecutionException;

public interface ProductService {

    List<Product> getProducts(String uid) throws ExecutionException, InterruptedException;
    String addProduct(Product product) throws ExecutionException, InterruptedException;
    String updateProduct(Product product) throws ExecutionException, InterruptedException;
    String deleteProduct(String name);
    Product getProduct(String name) throws ExecutionException, InterruptedException;
}

File: client/webtim/src/services/ProductService.js
import axios from 'axios';

const PRODUCTS_API_URL = "http://localhost:8080/api/";

class ProductService {

    getProducts(uid){
        return axios.get(PRODUCTS_API_URL + "products" + "/" + uid);
    }

    addProduct(product){
        return axios.post(PRODUCTS_API_URL + "add", product)
    }

    getProduct(name){
        return axios.get(PRODUCTS_API_URL + 'product/' + name)
    }

    deleteProduct(name){
        return axios.delete(PRODUCTS_API_URL + "delete/" + name)
    }

    updateProduct(product){
        return axios.put(PRODUCTS_API_URL + "update", product)
    }
}

export default new ProductService()